### DISCUSSION TITLE
Hi, guys, how the get the weights of inputs and outputs with DEA models?

### DISCUSSION THREAD

**Discussion by GithubRocky200** [NibuTake/PyDEA#22] [Q&A] [answered]
Hi, guys, how the get the weights of inputs and outputs with DEA models? It seems I cannot get them with the following code: print([r.weight for r in dea.result]). Is anyone know? Thank you very much.

**Comment by GithubRocky200** [comment]
I use Pyfrontier.frontier_model EnvelopDEA

**Reply by NibuTake** [reply, marked as answer]
@GithubRocky200 

Thank you for reporting and using our module!
Actually, the weight in the Envelopment model refers to the weight of the DMUs (often denoted as lambda). To prevent confusion in the future, we may change the term "weight" to "lambda."

If you'd like to get the weight of  x (inputs) and y (outputs), please use a multiple DEA model like this.

```python
import pandas as pd
from Pyfrontier.frontier_model import MultipleDEA

sample_df = pd.DataFrame(
    {"day": [1, 2, 4, 6, 4], "cost": [5, 2, 1, 1, 4], "profit": [15, 15, 15, 15, 15]}
)

dea = MultipleDEA("CRS", "in")
dea.fit(
    sample_df[["day", "cost"]].to_numpy(),
    sample_df[["profit"]].to_numpy(),
)

print([(r.x_weight, r.y_weight) for r in dea.result])
```

**Reply by GithubRocky200** [reply]
@NibuTake 
Thank you very much your quick reply! I have gotten the weight of inputs and outputs with your suggestions by using the MultipleDEA model. Thank you again.
